mindsdb/api/mysql/mysql_proxy/classes/sql_query.py

```diff
@@ -872,8 +870,8 @@ def execute_step(self, step, steps_data):
                     row = row[right_key]
                     right_df_data.append({right_columns_map_reverse[key]: value for key, value in row.items()})
 
-                df_a = pd.DataFrame(left_df_data)
-                df_b = pd.DataFrame(right_df_data)
+                df_a = pd.DataFrame(left_df_data, columns=left_columns_map.keys())
+                df_b = pd.DataFrame(right_df_data, columns=right_columns_map.keys())
```

Comment:
dict keys order is not strict in py3.6. We still need support it

Reply:
I will change to OrderedDict